packages/canon/src/components/DataTable/Pagination/types.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Table } from '@tanstack/react-table';
+
+/** @public */
+export interface DataTablePaginationProps<TData>
+  extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * The table instance.
+   */
+  table?: Table<TData>;
```

Comment:
hmm, it's a bit scary to have these tied directly to react-table, ofc also very powerful though.

Reply:
Just sharing the idea after talking about it - For now we would like to keep `Table` as plain table component and use `DataTable` as an opinionated version of Tanstack Table.